### ISSUE TITLE
3D data segmentation data transforms policies are inconsistent

### ISSUE THREAD

**Issue by KarlThelor** [Project-MONAI/tutorials#517]
**Describe the bug**

the transforms in [spleen_segmentation_3d.ipynb](https://github.com/Project-MONAI/tutorials/blob/master/3d_segmentation/spleen_segmentation_3d.ipynb) is that：
```python
train_transforms = Compose(
    [
        LoadImaged(keys=["image", "label"]),
        **EnsureChannelFirstd**(keys=["image", "label"]),
        Spacingd(keys=["image", "label"], pixdim=(
            1.5, 1.5, 2.0), mode=("bilinear", "nearest")),
        Orientationd(keys=["image", "label"], axcodes="RAS"),
        ScaleIntensityRanged(
            keys=["image"], a_min=-57, a_max=164,
            b_min=0.0, b_max=1.0, clip=True,
        ),
        CropForegroundd(keys=["image", "label"], source_key="image"),
        RandCropByPosNegLabeld(
            keys=["image", "label"],
            label_key="label",
            spatial_size=(96, 96, 96),
            pos=1,
            neg=1,
            num_samples=4,
            image_key="image",
            image_threshold=0,
        ),
        # user can also add other random transforms
        # RandAffined(
        #     keys=['image', 'label'],
        #     mode=('bilinear', 'nearest'),
        #     prob=1.0, spatial_size=(96, 96, 96),
        #     rotate_range=(0, 0, np.pi/15),
        #     scale_range=(0.1, 0.1, 0.1)),
        EnsureTyped(keys=["image", "label"]),
    ]
)

```
but in [tutorial spleen_segmentation_3d_lightning.ipynb ](https://github.com/Project-MONAI/tutorials/blob/master/3d_segmentation/spleen_segmentation_3d_lightning.ipynb) is that :

```python
 train_transforms = Compose(
            [
                LoadImaged(keys=["image", "label"]),
                **AddChanneld**(keys=["image", "label"]),
                Spacingd(
                    keys=["image", "label"],
                    pixdim=(1.5, 1.5, 2.0),
                    mode=("bilinear", "nearest"),
                ),
                Orientationd(keys=["image", "label"], axcodes="RAS"),
                ScaleIntensityRanged(
                    keys=["image"], a_min=-57, a_max=164,
                    b_min=0.0, b_max=1.0, clip=True,
                ),
                CropForegroundd(keys=["image", "label"], source_key="image"),
                # randomly crop out patch samples from
                # big image based on pos / neg ratio
                # the image centers of negative samples
                # must be in valid image area
                RandCropByPosNegLabeld(
                    keys=["image", "label"],
                    label_key="label",
                    spatial_size=(96, 96, 96),
                    pos=1,
                    neg=1,
                    num_samples=4,
                    image_key="image",
                    image_threshold=0,
                ),
                # user can also add other random transforms
                #                 RandAffined(
                #                     keys=['image', 'label'],
                #                     mode=('bilinear', 'nearest'),
                #                     prob=1.0,
                #                     spatial_size=(96, 96, 96),
                #                     rotate_range=(0, 0, np.pi/15),
                #                     scale_range=(0.1, 0.1, 0.1)),
                EnsureTyped(keys=["image", "label"]),
            ]
        )

```

after the process in  spleen_segmentation_3d_lightning.ipynb network will get input data of [1,1,96,96,96] shape, the last channel seemed to be depth.This does not  fit the data format that PyTorch [needs](https://pytorch.org/docs/stable/generated/torch.nn.Conv3d.html?highlight=conv3d#torch.nn.Conv3d)
![image](https://user-images.githubusercontent.com/60059677/150666911-cd32a2f6-beda-4587-ab70-0850656bb0d3.png)
Is there any impact if the number of channels does not match?

**Comment by Nic-Ma** [comment]
Hi @ericspod ,

I think in the medical domain, the `HWD` definition is slightly different from PyTorch description for 3D operations.
Could you please help share more details as you are expert on that?

Thanks in advance.

**Comment by KarlThelor** [comment]
Thank you for your reply！ I am a beginner,  the problem of channel position has caused some troubles to me. In another medical segmentation framework nnunet[](https://github.com/MIC-DKFZ/nnUNet)，Data is organized into NCHW and NCDHW formats as PyTorch's documents and sent to network training.
![nnunet 2d](https://user-images.githubusercontent.com/60059677/150680547-48626b26-df66-441f-a92a-41f130ff66e4.png)
![nnunet 3d](https://user-images.githubusercontent.com/60059677/150680559-349d711b-db04-471f-9c17-027584d890a1.png)

**Comment by Nic-Ma** [comment]
Hi @KarlThelor ,

Our transforms are `channel-first`, the data shape is `CHWD`, then `DataLoader` will batch data to `NCHWD` to network.
And actually, I think you don't need to worry about `HWD` or `DHW`, that depends on which spatial dim you want to define as the `D` of PyTorch 3D in medical images.

Thanks.
